pkg/image/imageutil/helpers.go

```diff
@@ -200,3 +204,135 @@ func PrioritizeTags(tags []string) {
 		tags[i] = pt.tag
 	}
 }
+
+// StatusHasTag returns named tag from image stream's status and boolean whether one was found.
+func StatusHasTag(stream *imagev1.ImageStream, name string) (imagev1.NamedTagEventList, bool) {
+	for _, tag := range stream.Status.Tags {
+		if tag.Tag == name {
+			return tag, true
+		}
+	}
+	return imagev1.NamedTagEventList{}, false
+}
+
+// LatestTaggedImage returns the most recent TagEvent for the specified image
+// repository and tag. Will resolve lookups for the empty tag. Returns nil
+// if tag isn't present in stream.status.tags.
+func LatestTaggedImage(stream *imagev1.ImageStream, tag string) *imagev1.TagEvent {
+	if len(tag) == 0 {
+		tag = imagev1.DefaultImageTag
+	}
+
+	// find the most recent tag event with an image reference
+	t, ok := StatusHasTag(stream, tag)
+	if ok {
+		if len(t.Items) == 0 {
+			return nil
+		}
+		return &t.Items[0]
+	}
+
+	return nil
+}
+
+// ImageWithMetadata mutates the given image. It parses raw DockerImageManifest data stored in the image and
+// fills its DockerImageMetadata and other fields.
+// Copied from github.com/openshift/image-registry/pkg/origin-common/util/util.go
+func ImageWithMetadata(image *imagev1.Image) error {
```

Comment:
Can we just return that `dockerv10.DockerImage` instead of filing it in back? That'd save us from those nasty casts we do all over the place?